node/src/components/consensus/protocols/highway/synchronizer/tests.rs

```diff
@@ -111,3 +108,128 @@ fn purge_vertices() {
+    let unit = |hash: u64| Vertex::Unit(state.wire_unit(&hash, TEST_INSTANCE_ID).unwrap());
+    // Returns the PreValidatedVertex with the specified hash.
+    let pvv = |hash: u64| util_highway.pre_validate_vertex(unit(hash)).unwrap();
+
+    let peer0 = NodeId(0);
+
+    // Create a synchronizer with a 0x20 ms timeout, and a Highway instance.
+    let mut sync = Synchronizer::<NodeId, TestContext>::new(
+        0x20.into(),
+        0x20.into(),
+        WEIGHTS.len(),
+        TEST_INSTANCE_ID,
+    );
+
+    let mut highway = Highway::<TestContext>::new(TEST_INSTANCE_ID, test_validators(), params);
+    let now = 0x20.into();
+
+    assert!(matches!(
+        *sync.schedule_add_vertex(peer0, pvv(c2), now),
+        [ProtocolOutcome::QueueAction(ACTION_ID_VERTEX)]
+    ));
+    // `c2` can't be added to the protocol state yet b/c it's missing its `c1` dependency.
+    let (pv, outcomes) = sync.pop_vertex_to_add(&highway);
+    assert!(pv.is_none());
+    assert_targeted_message(
+        &unwrap_single(outcomes),
+        &peer0,
+        HighwayMessage::RequestDependency(Dependency::Unit(c1)),
+    );
+    // Simulate `c1` being downloaded…
+    let c1_outcomes = sync.schedule_add_vertex(peer0, pvv(c1), now);
+    assert!(matches!(
+        *c1_outcomes,
+        [ProtocolOutcome::QueueAction(ACTION_ID_VERTEX)]
+    ));
+    // `c1` is now part of the synchronizer's state, we should not try requesting it if other
+    // vertices depend on it.
+    assert!(sync.schedule_add_vertex(peer0, pvv(b0), now).is_empty());
+    // `b0` can't be added to the protocol state b/c it's missing its `c1` dependency,
+    // but `c1` has already been downloaded so we should not request it again. We will only request
+    // `c0` as that's what `c1` depends on.
+    let (pv, outcomes) = sync.pop_vertex_to_add(&highway);
+    assert!(pv.is_none());
+    assert_targeted_message(
+        &unwrap_single(outcomes),
+        &peer0,
+        HighwayMessage::RequestDependency(Dependency::Unit(c0)),
+    );
+    // "Download" the last dependency.
+    let _ = sync.schedule_add_vertex(peer0, pvv(c0), now);
+    // Now, the whole chain can be added to the protocol state.
+    for unit in &[c0, c1, b0, c2] {
+        let (maybe_pv, outcomes) = sync.pop_vertex_to_add(&highway);
+        assert_eq!(Dependency::Unit(*unit), maybe_pv.unwrap().vertex().id());
+        // Verify that we don't request any dependency now.
+        assert!(
+            !outcomes
+                .iter()
+                .any(|outcome| matches!(outcome, ProtocolOutcome::CreatedTargetedMessage(_, _))),
+            "unexpected dependency request {:?}",
+            outcomes
+        );
+        let vv = highway
+            .validate_vertex(pvv(*unit))
+            .unwrap_or_else(|_| panic!("{:?} unit is valid", unit));
+        highway.add_valid_vertex(vv, now);
+        let _ = sync.remove_satisfied_deps(&highway);
+    }
+    assert!(sync.is_empty());
+}
+
+#[cfg(test)]
```

Comment:
Isn't that annotation on the whole `tests` module already? No need to repeat it on individual items.

Reply:
Removed in [`d3e8a82` (#1312)](https://github.com/CasperLabs/casper-node/pull/1312/commits/d3e8a82bae9fc19cf5a7839a640b0a219720e6c5).